### PR TITLE
CMSIS-NN: Add offset check for int16 kernels

### PR DESCRIPTION
**Pull request by hmogensen-arm**
Add check that input and output tensors with int16 have offset zero
BUG=Fixes https://github.com/tensorflow/tflite-micro/issues/1364

### CODE REVIEW

File: tensorflow/lite/micro/kernels/pooling_common.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+/* Copyright 2021-2022 The TensorFlow Authors. All Rights Reserved.
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/pooling_common.cc:1]: Please replace 2021 with 2022.

```
Copyright 2022
```

File: tensorflow/lite/micro/kernels/svdf_common.cc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+/* Copyright 2020, 2022 The TensorFlow Authors. All Rights Reserved.
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/svdf_common.cc:1]: make this `Copyright 2022`

File: tensorflow/lite/micro/kernels/svdf_common.cc [hunk 3]
```diff
@@ -487,6 +487,10 @@ TfLiteStatus PrepareSvdf(TfLiteContext* context, TfLiteNode* node) {
             &(data->scratch_output_tensor_index));
     TF_LITE_ENSURE_OK(context, scratch_output_status);
   } else {
+    if (input->type == kTfLiteInt16) {
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/svdf_common.cc:490]: SVDF does not support int16 activations. I would recommend leaving this file unchanged.

**Review by advaitjain — CHANGES_REQUESTED** [review]
Thanks for the PR! Can you please add a BUG=<link to issue> to the PR description?

File: tensorflow/lite/micro/kernels/svdf_common.cc (hunk 3 above)
Comment (advaitjain) [reply tensorflow/lite/micro/kernels/svdf_common.cc:490]: could you revert the changes in svdf_common? or if you think this is needed, please let me know why.

**Review by advaitjain — CHANGES_REQUESTED** [review]
for future, if a force-push can be avoided that would be great. It helps make sure we can review incremental changes.

What that means is that try not to `git rebase` but instead use `git merge`.

See this for more info: https://github.com/tensorflow/tflite-micro/blob/main/CONTRIBUTING.md#during-the-pr-review

File: tensorflow/lite/micro/kernels/svdf_common.cc (hunk 3 above)
Comment (hmogensen-arm) [reply tensorflow/lite/micro/kernels/svdf_common.cc:490]: My apologies, I was changing the wrong file. I will upload a new PR once it has passed the internal review

**Review by advaitjain — APPROVED** [review]

**Pull request merged**